src/google/protobuf/compiler/command_line_interface_unittest.cc

```diff
@@ -1779,6 +1779,39 @@ TEST_F(CommandLineInterfaceTest,
                     "$tmpdir/bar.pb: "
                     "$tmpdir/foo.proto\\\n $tmpdir/bar.proto");
 }
+
+TEST_F(CommandLineInterfaceTest,
+       DependencyManifestFileIsOrderedByFilePath) {
+  CreateTempFile("foo.proto",
+                 "syntax = \"proto2\";\n"
+                 "message Foo {}\n");
+  CreateTempFile("bar.proto",
+                 "syntax = \"proto2\";\n"
+                 "import \"foo.proto\";\n"
+                 "message Bar {\n"
+                 "  optional Foo foo = 1;\n"
+                 "}\n");
+
+  Run("protocol_compiler --dependency_out=$tmpdir/manifest_a "
+      "--test_out=$tmpdir "
+      "--descriptor_set_out=$tmpdir/a.pb --proto_path=$tmpdir bar.proto");
+
+  ExpectNoErrors();
+
+  ExpectFileContent("manifest_a",
+                    "$tmpdir/a.pb \\\n$tmpdir/bar.proto.MockCodeGenerator.test_generator: "
+                    "$tmpdir/foo.proto\\\n $tmpdir/bar.proto");
```

Comment:
Isn't this the wrong order?  I would expect bar to come before foo...

Reply:
I only sorted the outputs as they are the ones that cause the problems.
I could also sort the dependencies list but I don't think it's necessary

Reply:
gotcha, I'm not sure I see how this test locks down the new behavior then.  Am I missing something?

Reply:
Basically, which test assertion makes sure the outputs are sorted?

Reply:
So I do two `protoc` calls, one outputting a descriptor file with the name `a.pb` and the second time with `z.pb` while the test generator output is always `bar.proto.MockCodeGenerator.test_generator`

So one manifest file looks something like this (minus newlines and tempdir)
`a.pb bar.proto.MockCodeGenerator.test_generator: foo.proto bar.proto`
and the other like
`bar.proto.MockCodeGenerator.test_generator z.pb: foo.proto bar.proto`
having sorted the outputs alphabetically

Reply:
Ah thanks for explaining that!